hooks/pre-command

```diff
@@ -136,17 +153,50 @@ if [[ ! -d "${unpack_dest}" || "${orig_mtime}" != "${post_dl_mtime}" ]]; then
     echo "New or updated installation detected, unpacking!"
     rm -rf "${unpack_dest}"
     mkdir -p "${unpack_dest}"
-    tar -xof "${tarball_path}" --strip-components=1 -C "${unpack_dest}"
+    if [[ "${archive_path}" == *.tar.gz ]]; then
+      tar -xof "${archive_path}" --strip-components=1 -C "${unpack_dest}"
+    elif [[ "${archive_path}" == *.zip ]]; then
+      unzip -q "${archive_path}" -d "${unpack_dest}"
+      mv "${unpack_dest}"/julia-*/* "${unpack_dest}"
+      rmdir "${unpack_dest}"/julia-*
+    else
+      buildkite-agent annotate --style error "Unhandled archive type ${archive_path}"
+      exit 1
+    fi
 fi
-# Export `PATH` such that this `julia` is the first one other plugins find
-export PATH=${unpack_dest}/bin:$PATH
 
 # Show that julia is installed and print out the version info
 echo "--- :julia: Installation Successful"
-echo "Julia ${JULIA_VERSION} successfully installed to $(which julia)"
-julia -e 'using InteractiveUtils; versioninfo()'
+echo "Julia ${JULIA_VERSION} successfully installed to ${unpack_dest}"
+"${unpack_dest}/bin/julia" -e 'using InteractiveUtils; versioninfo()'
 
+# Export `PATH` such that this `julia` is the first one other plugins find
+if [[ "$OSTYPE" == "linux"* ]]; then
+  export PATH=${unpack_dest}/bin:"$PATH"
+elif [[ "$OSTYPE" == "msys" ]]; then
+  # modifying the environment on Windows doesn't work, not by setting the
+  # PATH env var (picked up by a Buildkite pre-exit hook, but subsequently
+  # setting an incomplete environment later on, failing to find bash.exe again),
+  # or even by directly modifying HKCU\Environment:Path (which somehow does not
+  # get picked up by subsequently-executed Buildkite plugins.
```

Comment:
Perhaps we can complain to the buildkite folks about this?  It seems like an important usecase.

Reply:
I'm not sure who is to blame, as part of the reason is how Git bash translates the Windows path to a POSIX one. But I guess it can't hurt.

Reply:
Are you running the agent from within a git bash session?  Can I see a failing build log where the environment gets messed up?

Reply:
Extension-less files get executed by buildkite using git bash (installed manually) ndeed.

Example of a failure: https://buildkite.com/julialang/cuda-dot-jl/builds/1504#3f7bab3a-9871-4b7f-8da5-018b67a4e292. As you can see, that only happens during execution of the second plugin, presumable because the change to PATH detected by buildkite's exit hook during execution of the first plugin (which contains entries like `/usr/bin`, but not the original path to git bash, `c:/Program Files/Git/bin`) as done here: https://github.com/buildkite/agent/blob/848a053b3131ebced93d829c3215b31bddd19f72/hook/scriptwrapper.go#L105-L131 resulted in an 'invalid' environment being restored during execution of the test plugin. I couldn't inspect the temporary files though, as BK removes them and doesn't debug log them.

Reply:
buildkite itself is started by nssm, so it's a legit Windows service, but it detects and uses git bash to execute plugins.

Reply:
Probably https://github.com/buildkite/agent/issues/872; I'll give this a go next week.

Reply:
Can you do `MSYS_NO_PATHCONV=1`? Or is that insufficient?

Reply:
IIUC that would need to be set in all downstream plugins. The other workaround only needs to be applied here, and seems to work too.